### DISCUSSION TITLE
更友好的文档

### DISCUSSION THREAD

**Discussion by D-Sketon** [theme-shoka-x/hexo-theme-shokaX#298] [General]
网上对shokax的吐槽
https://www.cnblogs.com/tfel-ypoc/p/18141158

**Comment by zkz098** [comment]
npm module 这个改不了，hexo 不会自动排除 ts 文件，所以script必须做预处理（话说这算不算上游 Bug，scripts中的文件不会自动排除无法运行的，即使JSON也会运行）
文档这个问题，等V2会有好转。
SXC本身地位就比较尴尬，其实本质目的就是处理对等依赖问题，`uninstall`确实是设计缺陷
PM这个参数其实 SXC install --help 里是有的，但文档里确实没有
yarn一堆报错那个我没遇到过，可能是因为我一直用的pnpm

那个"补充"实在无法认同，按这个作者说法90%的二次开发/分叉项目都是无意义的，二次开发的目的不就是解决问题吗（

**Reply by D-Sketon** [reply]
yarn报错可能是minify库里找不到依赖，这个我确实碰到过

**Reply by FidStyle** [reply]
yarn好像确实，我指定的npm

**Reply by zkz098** [reply]
> yarn报错可能是minify库里找不到依赖，这个我确实碰到过

那就只能移除minify-html了（
主流构建系统好像都不会压缩html，其实可以考虑移除html压缩支持

**Reply by FidStyle** [reply]
npm module和git clone区别在哪。
大部分主题是直接clone方便修改寻找，npm这个目录确实太深了，下策是执行脚本（分系统即可）创建软链接到theme文件夹，但可能会引发链式反应需要做说明

**Reply by zkz098** [reply]
> npm module和git clone区别在哪。 大部分主题是直接clone方便修改寻找，npm这个目录确实太深了，下策是执行脚本（分系统即可）创建软链接到theme文件夹，但可能会引发链式反应需要做说明

可以直接把SXC也连带合并入postinstall.js内，当然也是治标不治本
如果能确定scripts那个是上游问题，hexo修复后其实 git clone 也能用

**Reply by D-Sketon** [reply]
特别是如果使用pnpm并修改了node_modules里的主题，会污染整个全局主题

**Reply by zkz098** [reply]
有个方案：可以 git clone，但clone完需要自行编译（）
感觉并不比npm module 好多少

**Reply by FidStyle** [reply]
> 有个方案：可以 git clone，但clone完需要自行编译（） 感觉并不比npm module 好多少

可以给出，让用户选择即可

**Reply by FidStyle** [reply]
> 特别是如果使用pnpm并修改了node_modules里的主题，会污染整个全局主题

pnpm特性没办法，所以我只能选择最慢的npm（yarn用不了，pnpm硬链接不能局部修改）

**Reply by D-Sketon** [reply]
建议cli不要将minify作为绑定依赖 因为sharp因为网络原因安装失败的概率也很大

**Reply by zkz098** [reply]
https://github.com/theme-shoka-x/shokax-can
做了个罐头，V2文档中作为 0.4.5 以后的推荐安装方式

**Reply by zkz098** [reply]
V2文档已进入测试，目前不适用于生产环境: https://docs-hexo.shokax.top

**Reply by zkz098** [reply]
另外 ShokaX 是不是有必要整个用户群和开发群，有个群应该能优化一下相关情况（）

**Reply by D-Sketon** [reply]
> 另外 ShokaX 是不是有必要整个用户群和开发群，有个群应该能优化一下相关情况（）

可以考虑

**Reply by zkz098** [reply]
Toolbox目前已制作完毕，允许从git clone后自行编译js

**Reply by zkz098** [reply]
postinstall已经完成，0.4.6准备进行测试

**Reply by zkz098** [reply]
需要协助，这个postinstall 安装时无法正常安装，手动调用Npm内的也不行，但是手动调用 git clone方式的可以
怪

**Reply by zkz098** [reply]
> 需要协助，这个postinstall 安装时无法正常安装，手动调用Npm内的也不行，但是手动调用 git clone方式的可以 怪

坏消息：postinstall没有那么post，所以无法在其中调用包管理器

**Reply by zkz098** [reply]
> > 需要协助，这个postinstall 安装时无法正常安装，手动调用Npm内的也不行，但是手动调用 git clone方式的可以 怪
> 
> 坏消息：postinstall没有那么post，所以无法在其中调用包管理器

好消息：换成install就能运行了
坏消息:  install script好像会被偶发性无视，不知道是不是仅pnpm存在此问题

**Reply by zkz098** [reply]
彻底被Hexo这个诡异的依赖问题整麻了
这个问题修差不多之后就准备把大部分精力放在 Astro 迁移上了，起码 Astro / Valaxy 这种现代系统不需要为依赖操心

**Reply by zkz098** [reply]
进入死循环了（）
编译脚本会自动从远程拉取 npm 注册表安装依赖，pnpm 安装时默认 save
~~这导致现在事实上地无法删除任何依赖（）~~

通过使用npm解决

**Reply by zkz098** [reply]
@D-Sketon @FidStyle 有时间测一下0.4.7的依赖安装正不正常，pnpm需要额外运行`node .\node_modules\hexo-theme-shokax\toolbox\hoistdep.mjs`
需要确定hexo-lightning-minify会不会自动安装，我这边环境可能已经被污染了

**Reply by FidStyle** [reply]
> @D-Sketon @FidStyle 有时间测一下0.4.7的依赖安装正不正常，pnpm需要额外运行`node .\node_modules\hexo-theme-shokax\toolbox\hoistdep.mjs` 需要确定hexo-lightning-minify会不会自动安装，我这边环境可能已经被污染了

`node .\node_modules\hexo-theme-shokax\toolbox\hoistdep.mjs`后没发现hexo-lightning-minify对应包

**Comment by zkz098** [comment]
`uninstall`那个好像是普遍缺陷，其实大部分需要前置的 hexo 主题都没有这个东西（）
如果后续postinstall、自编译和即食罐头表现良好，SXC就准备弃用了
